### DISCUSSION TITLE
Hazel tunnel integration in camoufox?

### DISCUSSION THREAD

**Discussion by desoforgit** [daijro/camoufox#27] [General]
I remember it was planned but is no longer mentioned in the 'What's planned?' section.
Any particular reason for it?

**Comment by daijro** [comment]
Hello,

Since only Firefox fingerprints are planned to be supported by Camoufox, I didn't see a reason anymore to add a functionality that spoofs the TLS fingerprint to other browsers.

Introducing other browsers like Chrome cause detection issues, since its impossible to spoof Firefox's JS engine. And recently, more WAFs are starting to detect if the Firefox version has been spoofed (#14).

In order to keep Camoufox as undetectable as possible, I chose to not allow spoofing of the browser or version. I don't think it's reliable or possible. Instead, I think a different path for this project could be making a local docker server and web interface to remotely connect to and rotate between different versions of Camoufox instead of spoofing the Firefox version.
